spec/index.adoc

```diff
@@ -3134,6 +3150,7 @@ S with
       controllers[A.canister_id] = A.settings.controllers
     if A.settings.freezing_threshold is not null:
       freezing_threshold[A.canister_id] = A.settings.freezing_threshold
+    canister_state_counter[A.canister_id] = S.canister_state_counter[A.canister_id] + 1
```

Comment:
Let's just increment the state counter for every change of the canister's settings unless it's easier to increment it only for changes of its controllers.